File: src/main/java/com/codeperfection/shipit/dto/shipping/ShippedItemDto.java
package com.codeperfection.shipit.dto.shipping;

import com.codeperfection.shipit.dto.product.ProductDto;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.UUID;

@Data
@AllArgsConstructor
public class ShippedItemDto {

    private UUID uuid;

    private ProductDto product;

    private Integer count;
}

File: src/main/kotlin/com/codeperfection/shipit/dto/shipping/ShippedItemDto.kt
package com.codeperfection.shipit.dto.shipping

import com.codeperfection.shipit.dto.product.ProductDto
import com.codeperfection.shipit.entity.ShippedItem
import java.util.UUID

data class ShippedItemDto(
    val id: UUID,
    val product: ProductDto,
    val quantity: Int
) {

    companion object {

        fun fromEntity(shippedItem: ShippedItem) = ShippedItemDto(
            id = shippedItem.id,
            product = ProductDto.fromEntity(shippedItem.product),
            quantity = shippedItem.quantity
        )
    }
}
